src/NServiceBus.Core/Transports/Msmq/MsmqUtilities.cs

```diff
@@ -231,9 +231,47 @@ static void AssignMsmqNativeCorrelationId(OutgoingMessage message, Message resul
             }
         }
 
+        public static bool TryOpenQueue(MsmqAddress msmqAddress, out MessageQueue messageQueue)
+        {
+            messageQueue = null;
+
+            var queuePath = msmqAddress.PathWithoutPrefix;
+
+            if (msmqAddress.IsRemote)
+            {
+                Logger.Debug("Queue is on remote machine.");
+                Logger.Debug("If this does not succeed (like if the remote machine is disconnected), processing will continue.");
+            }
+
+            Logger.Debug($"Checking if queue exists: {queuePath}.");
```

Comment:
I know you got this from the existing implementation, but I think this logging statement should come before the `IsRemote` check. Looking through the logs after the fact, it would make more sense.